website/content/specification/syntax/constraints.md

```diff
@@ -14,27 +14,79 @@ TODO: P3: Address issue https://github.com/usnistgov/metaschema/issues/325
 
 ## Enumerated values
 
-Additionally, flags may be constrained to a set of known values listed in advance.
+The `allowed-values` constraint is a kind of Metaschema constraint that restricts field or flag value(s) based on an enumerated set of permitted values.
 
-This restriction can be either:
+Metaschema processors MUST process `allowed-values` enumerations.
 
-1. strict (values must be in the list for document validity with `allow-other="no"` attribute for an `allowed-values` element) or
-2. loose (i.e. for documentation only, no effect in schemas, with `allow-other="yes"`).
+The `@target` attribute of an `<allowed-values>` constraint is used to determine the specific content values the constraint applies to. By evaluating the `@target`, it is possible to identify the *target set* of `<allowed-values>` constraints targeting a given value.
 
-If an `allowed-values` constraint does not have the `allow-other` attribute defined, the default is `allow-other="no"`, resulting in strict validation where the only valid values are those in the list.
+The *source* of each allowed `<allowed-values>` constraint in the *target set* is identified as one of the following sources:
 
-Within `allowed-values` of a `constraint`, an `enum` element's `@value` attribute assigns the permissible value, while its data content provides documentation. For example:
+- **model:** The constraint is defined *in* a Metaschema module.
+- **external:** The constraint is defined *outside* a Metaschema module, e.g. external constraints.
+
+The *target set* of `<allowed-values>` constraints is verified for correctness using the `@extension` attribute on each set member.
+
+Once the *target set* of `<allowed-values>` constraints is validated, the `@allow-other` attribute is used to determine the *expected value set* for a given content value.
```

Comment:
But isn't there only one `allow-other` attribute but per `allowed-values` constraint, not attributes plural?

Reply:
It is a set of constraints, which apply to a given node. There could be many such constraints, many `allow-other` constraints, and many values of `allow-other` given on those. This is why "respective".

It could be rephrased as "the `@allow-other` attribute (or defaulted value) for each applicable `allow-other` constraint" or some such.

The point is that the target is not the constraint, it is the node to which a constraint applies. Each constraint has multiple targets, and (now, under these rules), each node can have multiple targeting constraints, not only each one applicable, but applicable collectively.